sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_services_first_seen_v1/init.sql

```diff
@@ -1,144 +1,18 @@
-    w1_unframed AS (
-      PARTITION BY
-        user_id,
-        service
-      ORDER BY
-        `timestamp`
-    )
-),
-  -- we need this next section because `did_register` will be BOTH true and false within the flows that the user registered on.
-  -- this dedupes the rows from above and sets did_register to true only on flows that included a registration
-  -- I've verified that `date(first_service_timestamp), count(distinct user_id) where did_register = true group by 1`  matches the counts of registrations per day in amplitude.
-first_services_g AS (
-  SELECT
-    * EXCEPT (_n)
-  FROM
-    first_services
-  WHERE
-    _n = 1
-),
-  -- sadly, `entrypoint` is null on registration complete and login complete events.
-  -- this means we have to use first_service_flow to join back on the original source table's flow_id,
-  -- and take the first occurrence of `entrypoint` within the flow that the user first appeared in the service on.
-flows AS (
-  SELECT DISTINCT
-    s.first_service_flow,
-    FIRST_VALUE(f.entrypoint) OVER (
-      PARTITION BY
-        f.flow_id
-      ORDER BY
-        f.`timestamp`
-      ROWS BETWEEN
-        UNBOUNDED PRECEDING
-        AND UNBOUNDED FOLLOWING
-    ) AS first_service_entrypoint,
-    FIRST_VALUE(f.timestamp) OVER (
-      PARTITION BY
-        f.flow_id
-      ORDER BY
-        f.`timestamp`
-      ROWS BETWEEN
-        UNBOUNDED PRECEDING
-        AND UNBOUNDED FOLLOWING
-    ) AS first_service_timestamp,
-    FIRST_VALUE(f.country) OVER (
-      PARTITION BY
-        f.flow_id
-      ORDER BY
-        f.`timestamp`
-      ROWS BETWEEN
-        UNBOUNDED PRECEDING
-        AND UNBOUNDED FOLLOWING
-    ) AS first_service_country,
-    FIRST_VALUE(f.os_name) OVER (
-      PARTITION BY
-        f.flow_id
-      ORDER BY
-        f.`timestamp`
-      ROWS BETWEEN
-        UNBOUNDED PRECEDING
-        AND UNBOUNDED FOLLOWING
-    ) AS first_service_os
-  FROM
-    first_services_g s
-  INNER JOIN
-    firefox_accounts.fxa_content_auth_oauth_events AS f
-  ON
-    s.first_service_flow = f.flow_id
-  WHERE
-    f.entrypoint IS NOT NULL
-    AND s.first_service_flow IS NOT NULL
-    AND DATE(f.`timestamp`) >= '2019-03-01'
-)
-  -- finally take the entrypoint data and join it back on the other information (os, country etc).
-  -- also, add a row number that indicates the order in which the user signed up for their services.
-SELECT
-  s.user_id,
-  s.service,
-  s.first_service_flow,
-  s.did_register,
-  f.first_service_entrypoint AS entrypoint,
-  f.first_service_timestamp,
-  f.first_service_country,
-  f.first_service_os,
-  ROW_NUMBER() OVER (PARTITION BY s.user_id ORDER BY first_service_timestamp_last) AS service_number
-FROM
-  first_services_g s
-LEFT JOIN
-  flows f
-USING
-  (first_service_flow)
+SELECT DISTINCT
+  CAST(NULL AS STRING) AS user_id,
+  CAST(NULL AS STRING) AS service,
+  CAST(NULL AS STRING) AS os_name,
+  CAST(NULL AS DATE) AS first_seen_date,
+FROM fxa_events
```

Comment:
@kik-kik are there additional dimensions we can add to this table that provide some useful information for first seen date?

Reply:
I'm considering adding flow_id so that we could use it to map a specific user to dimensions like entrypoint and country instead of adding this logic here.

Reply:
I think the flow_id is a good ide but I don't think using it to pull commonly used dimensions in will be very performant, especially in looker. I looked at the lineage for this table and we'll likely have a user_id with a 2019 start date. If we setup the model to look for the country for this client, then we'll have to query the `fxa_content_auth_oauth_events` view which is based on `fxa_events_v1`(4.85TB) and `fxa_auth_events_v1` (52TB) derived tables. I think it will be very slow. What do you think about pulling in what we think might be commonly used dimensions then including flow_id so that if an analyst wants to look at something that isn't pre aggregated then they have a much simpler join to make?

Reply:
let's chat about it next time we meet.